File: packages/flutter/lib/language/enum/value_object/language_code.dart
import 'package:allowance_questboard/core/messages/locale_string.dart';
import 'package:allowance_questboard/core/value_object/base_state_object.dart' show BaseValueObject;

/// 言語コードの値オブジェクト
class LanguageCode extends BaseValueObject<String> {
  LanguageCode(super.value);
  
  @override
  void validate() {
    validator.required();
    validator.integer(); // 言語コードは数値文字列
  }
  
  @override
  LocaleString get valueName => const LocaleString(
    ja: "言語コード",
    en: "Language Code",
  );
}

File: packages/shared/features/language/value-object/languageCode.ts
import { LocaleString } from '../../../core/messages/localeString';
import { BaseValueObject } from '../../../core/value-object/baseValueObject';

/**
 * 言語コードの値オブジェクト
 * PythonのLanguageCodeクラスのTypeScript版
 */
export class LanguageCode extends BaseValueObject<string> {
  constructor(value: string) {
    super({ value });
  }

  protected validate(): void {
    this.validator.required();
  }

  protected get valueName(): LocaleString {
    return new LocaleString({
      ja: "言語コード",
      en: "Language Code"
    });
  }
}
